commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/localization/LocalizationServiceImp.java

```
@@ -0,0 +1,124 @@
+            Locale locale = localePolicy.determineLocale(localeCtx);
+            lang = locale.getLanguage();
+
+        } catch (Exception e) {
+            LOG.error("Cannot get the current language of the user " + currentUser.getUserId());
+        } finally {
+            return  lang;
+        }
+
+    }
+
+
+    private Locale checkPortalLocaleSupported(Locale portalLocale, Set<Locale> supportedLocales) {
+
+        if (supportedLocales.contains(portalLocale))
+            return portalLocale;
+        if ("".equals(portalLocale.getCountry()) == false) {
+            Locale loc = new Locale(portalLocale.getLanguage());
+            if (supportedLocales.contains(loc)) {
+                LOG.warn("portalLocale not supported: " + LocaleContextInfo.getLocaleAsString(portalLocale)
+                        + ". Falling back to '" + portalLocale.getLanguage() + "'.");
+                portalLocale = loc;
+                return loc;
+            }
+        }
+
+        LOG.warn("portalLocale not supported: " + LocaleContextInfo.getLocaleAsString(portalLocale)
+                + ". Falling back to Locale.ENGLISH.");
+        portalLocale = Locale.ENGLISH;
+        return portalLocale;
+    }
+
+    private Locale getUserProfileLocale( ExoContainer container, String user) throws Exception {
+
+        UserProfile userProfile = null;
+        OrganizationService svc = (OrganizationService) container.getComponentInstanceOfType(OrganizationService.class);
+        if (user != null) {
+            try {
+                beginContext(svc);
+                userProfile = svc.getUserProfileHandler().findUserProfileByName(user);
+            } catch (Exception ignored) {
+                LOG.error("IGNORED: Failed to load UserProfile for username: " + user, ignored);
+            } finally {
+                try {
+                    endContext(svc);
+                } catch (Exception ignored) {
+                    // we don't care
```

Never swallow exception. At least log it.